umbraco/Umbraco-CMS
# V8: Fix "ambient scope" error when fetching macros by Guid

### Prerequisites


If there's an existing issue for this PR then this fixes https://github.com/umbraco/Umbraco-CMS/issues/5956

### Description

LINQ lazy execution causes the error described in #5956... this PR fixes it by evaluating the LINQ statement explicitly before returning.

#### Steps to reproduce/test

1. Create a macro and grab its GUID:
![image](https://user-images.githubusercontent.com/7405322/61563296-d4d1f000-aa73-11e9-9d78-1c25e74126d6.png)
2. Render any piece of content with this template, replacing "YOUR MACRO GUID HERE" with the actual macro GUID:
```cshtml
@inherits Umbraco.Web.Mvc.UmbracoViewPage
@{
    Layout = null;
	
    var guid = new Guid("YOUR MACRO GUID HERE");
    var macros = Services.MacroService.GetAll(guid).ToArray();
}
<html>
    <body>
        <h1>IT BLOODY WORKS!</h1>
        <ul>
            @foreach(var macro in macros) {
                <li>@macro.Name</li>
            }
        </ul>
    </body>
</html>
```
3. If things don't blow up, the PR works 😄 and you should see the name of your macro rendered.
